### ISSUE TITLE
Add RZDB_KP_PROTOCOL

### ISSUE THREAD

**Issue by ZevEisenberg** [Raizlabs/RZDataBinding#49]
I'd like to add this:

```objc
#if DEBUG
#define RZDB_KP_PROTOCOL(ProtocolName, keypath) ({\
id<ProtocolName> _rzdb_keypath_obj; \
__unused __typeof(_rzdb_keypath_obj.keypath) _rzdb_keypath_prop; \
@#keypath; \
})
#else
#define RZDB_KP_PROTOCOL(ProtocolName, keypath) (@#keypath)
#endif
```

I don't know if this belongs in RZDB directly, but it came up recently in a project that's using RZDB anyway, so I already have access to the other key path macros:

```objc
+ (void)abc_importResponse:(id)responseObject inContext:(NSManagedObjectContext *)context
{
    [self rzi_objectsFromArray:responseObject inContext:context];
    if ([self conformsToProtocol:@protocol(ABCDeletableStoredObject)]) {
        [self rzv_deleteAllWhere:[NSPredicate predicateWithFormat:@"%K == %@", RZDB_KP_PROTOCOL(ABCDeletableStoredObject, flaggedForDelete), @(YES)] inContext:context];
    }
}
```

**Comment by ZevEisenberg** [comment]
One annoying caveat is that you don't get autocomplete on `flaggedForDelete` when typing `RZDB_KP_PROTOCOL(ABCDeletableStoredObject, flaggedForDelete)` in that example, but I think it's fine because you still get compile-time checking if the protocol changes.

**Comment by jvisenti** [comment]
Implemented in v2.0.3
